sql/core/src/test/scala/org/apache/spark/sql/execution/streaming/state/MapStateSuite.scala

```diff
@@ -67,6 +67,7 @@ class MapStateSuite extends StateVariableSuiteBase {
       assert(!testState.exists())
       assert(testState.getMap().hasNext === false)
     }
+    ImplicitGroupingKeyTracker.removeImplicitKey()
```

Comment:
Maybe just do this in setup (before) or teardown (after) in StateVariableSuiteBase, to ensure the cleanup is guaranteed to be done - as failure in test A won't clean up the thread local and introduce another failure. We do like to avoid cascading failures.

Reply:
Done! Thanks Jungtaek!